Issue 360 385

technabors opened the pull request:
Closes #360 closes #385 

Issue 385 -  had been fixed previously and works when you run the project locally but doesn't display correctly on Design.rax.io. I added `!important` to the line of code responsible for the font-weight change it hopes that will remedy once deployed

Issue 360 - Removed buttons from eyebrow, removed scss for eyebrow buttons, segmented `{{page.title}}` and the submit button into a nest 9/3 column. Added 1 line of css to center-align button contents. Changed  button from small to regular because it was closer to the mock up provided.

technabors commented:
Before:
![image](https://user-images.githubusercontent.com/586682/33390487-9f450afc-d4fb-11e7-8251-01598704ccff.png)

After: 
![image](https://user-images.githubusercontent.com/586682/33390495-a43dd976-d4fb-11e7-92c3-bf186790537f.png)

araiford commented on templates/component.html:
```diff
@@ -6,10 +6,13 @@
             <div class="hxCol-8">
                 <div class="hxRow">
-                    <div class="hxCol">
+                    <div class="hxCol-9">
```
Avoid using the hx grid for all layout needs. 

hxRows add extra bottom margin that throws off desired layouts when nested, and it makes the markup more complex.   Instead, try using more traditional css tactics, like floating elements to get the desired layout.

**Also,** there is a bug in the hxGrid that causes buttons to stretch to fill 100% of the horizontal space of the parent hxCol.  To overcome this we have to break the flex inheritance.

**Here's a screenshot highlighting these issues:**

![image](https://user-images.githubusercontent.com/933210/33394756-f70d3c8e-d508-11e7-9406-48f35948ecc1.png)

**To fix these issues, consider this markup:**

![image](https://user-images.githubusercontent.com/933210/33395054-f4aa3658-d509-11e7-95d8-716f4934e79e.png)

This removes the nested hxRow and columns, and adds a dummy div around the title, button and blurb.  The dummy div breaks the flex inheritance that causes the button to stretch.

Consider adding a .pull-right class to the _utilities.scss.

**Here's what it looks like after the change:**

![image](https://user-images.githubusercontent.com/933210/33395121-223c26ee-d50a-11e7-9a20-9dcb4bf4fa8d.png)

araiford commented:
the button should be title case `Submit Issue`

araiford commented:
Bah -- I also forgot to add to take the hxRow/hxCols out of the breadcrumb header.  They're also adding extra, un-needed vertical space.

nimbinatus commented:
Leaving a note before review:
@araiford @technabors, when https://github.com/rackerlabs/helix-ui/pull/92 appears in the production toolkit, some of the issues with rows and columns from the Helix grid will go away or be easily fixed (for example, adding the `hxGutterless` class to get rid of the extra whitespace). No one has tested adding the `hxSpan` classes to a paragraph tag just yet, either, but I imagine it would work if we just needed the spacing without the whole grid...

technabors commented:
@araiford See updates in 1cc726a. Quick notes:

- Added helper classes for floating, clearing, and inline-blocking to utilities
- `Utilities.scss`added to `website.scss` build task.
- Removed`hxCol`'s from `{{page.title}}` as well as the breadcrumbs in the header.

araiford commented:
LGTM!

technabors commented on assets/css/website.scss:
```diff
@@ -39,5 +39,6 @@ $bold-weight:    700;
 "website/status-label",
-"website/shame"
+"website/shame",
+"website/utilities"
```
Rolled back. Looks like I was wrong, and it parses fine in the original location.

technabors commented on templates/component.html:
```diff
@@ -5,12 +5,11 @@
-                        <h1 class="hxPageTitle">{{page.title}}</h1>
-                        <div class="component-blurb">{{page.usage}}</div>
-                    </div>
+                <div class="inline-block">
```
Agreed. Should we address this in this PR or merge and look as part of the larger 'make the site responsive' issue?

araiford replied:
This can probably be handled with the current responsive classes in Helix, by tinkering with the column sizing classes, and placing them on the hxCols in the header.

You could consider setting the thumbnail and title columns to 12 at smaller screen sizes, so they stack, instead of wrapping.
